### ISSUE TITLE
[Accessibility]: Y-axis visible false, X-axis first label overlap hidden 

### ISSUE THREAD

**Issue by boopalachandran-ibm** [carbon-design-system/carbon-charts#1793]
### Application/Team

ISL/SCIS

### Environment

Browser

### What happened? What did you expect to happen instead?

issue case: When we are making Y-axis visible: false, X-axis first coordinate label getting hidden as attached in screenshot.
expected: it should visible even if it is very big coordinate label. 
![image](https://github.com/carbon-design-system/carbon-charts/assets/167304430/b9c6484f-2730-4f57-8fcb-17c6bcb4a596)

### What WCAG 2.1 checkpoint does the issue violate?

WCAG 2.0

### Version

    "@carbon/charts": "^1.13.8",
    "@carbon/charts-angular": "^1.8.1",

### Data & options used

```js
data:
[
  {
    "group": "orderLineChart",
    "x-axis-key": "Wed Jun 23 2021",
    "y-axis-key": "24940.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Thu Jul 15 2021",
    "y-axis-key": "1113.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Tue Jul 06 2021",
    "y-axis-key": "1350.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Thu Jul 22 2021",
    "y-axis-key": "484.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Fri Aug 06 2021",
    "y-axis-key": "2478.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Fri Jun 25 2021",
    "y-axis-key": "1468.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Sat Jul 24 2021",
    "y-axis-key": "176.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Tue Aug 10 2021",
    "y-axis-key": "1907.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Wed Aug 11 2021",
    "y-axis-key": "1495.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  },
  {
    "group": "orderLineChart",
    "x-axis-key": "Wed Jun 30 2021",
    "y-axis-key": "732.0",
    "tooltipsConfig": {
      "name": "plannedShipDate"
    }
  }
]

options:
 {
  title: 'Line (time series)',
  axes: {
      "bottom": {
        "visible": true,
        "title": "Shipment date",
        "mapsTo": "x-axis-key",
        "scaleType": "labels"
      },
      "left": {
        "visible": false,
        "title": "Quantity",
        "mapsTo": "y-axis-key",
        "scaleType": "linear"
      },
      "right": {
        "mapsTo": "y-axis-key",
        "visible": false,
        "scaleType": "linear"
      },
      "top": {
        "mapsTo": "x-axis-key",
        "visible": false,
        "scaleType": "labels"
      }
  },
  curve: 'curveMonotoneX',
  height: '400px'
}
```

### StackBlitz repro

https://stackblitz.com/edit/stackblitz-starters-d3ifby?description=An%20angular-cli%20project%20based%20on%20@angular/animations,%20@angular/common,%20@angular/compiler,%20@angular/core,%20@angular/forms,%20@angular/platform-browser,%20@angular/platform-browser-dynamic,%20@angular/router,%20core-js,%20rxjs,%20tslib%20and%20zone.js&file=package.json,src%2Fdata.ts,src%2Fmain.ts,src%2Foptions.ts,src%2Fglobal_styles.css,src%2Findex.html&template=node&title=Angular%20Starter

### What priority level would this issue be in your opinion?

P2
